### DISCUSSION TITLE
Middleware Nested Directory detection ?

### DISCUSSION THREAD

**Discussion by vasilistotskas** [nuxt/nuxt#19615] [Questions]
Hello, i would like to split my route middlewares in folders under **middleware** directory. For example this: 

![Screenshot_1](https://user-images.githubusercontent.com/60323306/224545625-53ddc88b-92e4-41a0-8855-2db65b39208a.png)

To this:
![Screenshot_2](https://user-images.githubusercontent.com/60323306/224545636-ba332dac-fd37-4222-a930-82293ef0c464.png)

I have added in the nuxt.config.ts
```
imports: {
		dirs: ['middleware/**']
	}
```
but it does not working or i dont know how to use the middlewares in definePageMeta middleware.

**Comment by Baroshem** [comment]
Hey,

I think that you do not need to register imports like this.

You should be able to register middlewares like this -> https://nuxt.com/docs/guide/directory-structure/middleware#example-a-named-route-middleware

**Reply by vasilistotskas** [reply]
Yes i know but if i end up creating more middlewares the structure will be ugly
